d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -316,6 +319,27 @@ public void shutdown()
     });
   }
 
+  private ScheduledFuture<?> checkShutdownAndSchedule(Runnable runnable, long delay, TimeUnit unit) {
+    if (_executorService.isShutdown())
+    {
+      _log.warn("Attempting to schedule a task after _executorService was shutdown, will do nothing");
+      return null;
```

Comment:
Reviewers: is this safe? Unlike `execute()`, `executorservice.schedule` returns a `ScheduledFuture<?>` and so we need to return **something** here.

Reply:
https://github.com/linkedin/rest.li/pull/1065#discussion_r2093823724